components/brave_wallet/common/brave_wallet.mojom

```diff
@@ -404,11 +405,11 @@ interface EthTxController {
   // This returns different data depending on which network is currently selected in EthJsonRpcController
   GetAllTransactionInfo(string from) => (array<TransactionInfo> transaction_infos);
   AddObserver(pending_remote<EthTxControllerObserver> observer);
-  ApproveHardwareTransaction(string tx_meta_id) => (bool success, string message);
-  ProcessLedgerSignature(string tx_meta_id, string v, string r, string s) => (bool status);
-
   SpeedupOrCancelTransaction(string tx_meta_id, bool cancel) => (bool success, string tx_meta_id, string error_message);
   RetryTransaction(string tx_meta_id) => (bool success, string tx_meta_id, string error_message);
+  GetNonceForHardwareTransaction(string tx_meta_id) => (string nonce);
+  GetTransactionMessageToSign(string tx_meta_id) => (string message);
```

Comment:
these should at least return a bool for success/failure and an optional string. It would be better to have some kind of error enum to provide more information why it failed

Reply:
changed to optional